topiary/predictor.py

```diff
@@ -159,26 +228,123 @@ def epitopes_from_mutation_effects(
         #    around each mutation
         # 2) dictionary mapping varcode effect to start offset of subsequence
         #    within the full mutant protein sequence
-        protein_subsequences, protein_subsequence_offsets = \
+        effect_to_subsequence_dict, effect_to_offset_dict = \
             protein_subsequences_around_mutations(
                 effects=top_effects,
                 padding_around_mutation=self.padding_around_mutation)
 
-        binding_predictions = self.mhc_model.predict(protein_subsequences)
-        logging.info("MHC predictor returned %s peptide binding predictions" % (
-            len(binding_predictions)))
-        epitopes = build_epitope_collection_from_binding_predictions(
-            binding_predictions=binding_predictions,
-            protein_subsequences=protein_subsequences,
-            protein_subsequence_start_offsets=protein_subsequence_offsets,
-            wildtype_ligandome_dict=self.wildtype_ligandome_dict)
-        return apply_epitope_filters(
-            epitopes,
-            ic50_cutoff=self.ic50_cutoff,
-            percentile_cutoff=self.percentile_cutoff,
-            only_novel_epitopes=self.only_novel_epitopes)
-
-    def epitopes_from_variants(
+        # since we know that each set of variant effects has been
+        # reduced to a single 'top priority' effect, we can uniquely
+        # identify each variant sequence by its original genomic variant
+        variant_string_to_effect_dict = {
+            effect.variant.short_description: effect
+            for effect in effect_to_subsequence_dict.keys()
+        }
+        variant_string_to_subsequence_dict = {
+            effect.variant.short_description: seq
+            for (effect, seq) in effect_to_subsequence_dict.items()
+        }
+        variant_string_to_offset_dict = {
+            effect.variant.short_description: subseq_offset
+            for (effect, subseq_offset) in effect_to_offset_dict.items()
+        }
+        df = self.predict_named_sequences(variant_string_to_subsequence_dict)
+        logging.info("MHC predictor returned %d peptide binding predictions" % (
+            len(df)))
+
+        # since we used variant descrptions as the name of each sequence
+        # let's rename that column to be more informative
+        df = df.rename(columns={"source_sequence_name": "variant"})
+
+        # adjust offset to be relative to start of protein, rather
+        # than whatever subsequence we used for prediction
+        def compute_peptide_offset_relative_to_protein(row):
```

Comment:
Using `peptide_offset` here to then generate a `peptide_offset` seems error-prone. Can you call the new `peptide_offset` by a new name?

Reply:
I keep making that field longer, it was just `offset` before...

How about `peptide_offset_in_protein`?

Reply:
👍